### PR TITLE
HRCPP-421 Near cache eviction test and misc fixes

### PR DESCRIPTION
**Pull request by rigazilla**
This PR contains:

a test for the near cache evition
and some fixes on near cache tests

https://issues.jboss.org/browse/HRCPP-421

### CODE REVIEW

File: xunit-test/NearCacheTest/CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ target_link_libraries(NearCacheMultiClientTest
 add_test (start_server ${PYTHON_EXECUTABLE} ${CMAKE_ROOT_SOURCE_DIR}/test/bin/server_ctl.py start ${JAVA_RUNTIME} ${HOTROD_JBOSS_HOME} standalone.xml)
 add_test (probe_port ${PYTHON_EXECUTABLE} ${CMAKE_ROOT_SOURCE_DIR}/test/bin/probe_port.py localhost 11222 60)
 
-add_test(xunit_nearCacheMultiClientTest NearCacheMultiClientTest --gtest_output=xml:${PROJECT_BINARY_DIR}/test-output/)
+add_test(xunit_nearCacheTest xunit_nearCacheTest --gtest_output=xml:${PROJECT_BINARY_DIR}/test-output/)
 
 add_test (stop_server ${PYTHON_EXECUTABLE} ${CMAKE_ROOT_SOURCE_DIR}/test/bin/server_ctl.py stop)
 add_test (probe_port_stop ${PYTHON_EXECUTABLE} ${CMAKE_ROOT_SOURCE_DIR}/test/bin/probe_port.py localhost 11222 60 down)
```
Comment (alanfx) [inline xunit-test/NearCacheTest/CMakeLists.txt]: `NearCacheEvictionTest` is not being executed

Comment (rigazilla) [reply xunit-test/NearCacheTest/CMakeLists.txt]: @alanfx the tests fail randomically, I'm looking into.

**Comment by rigazilla** [comment]
Changed to unsigned type the size of the near cache
fixed bug on near cache entry remove
added more robust checks in the test suite

@alanfx  be sure to rebuild the project from scratch, the new test needs new makefile from cmake

**Comment by alanfx** [comment]
@rigazilla merging. More tests like failover can be added once the framework supports it.

**Pull request merged**